### DISCUSSION TITLE
Timing / context

### DISCUSSION THREAD

**Discussion by nashjc** [SudoWodo/StochOptim#16] [General]
Somewhere when an optimizer is run on a particular problem we should time the execution.
For now, system.time() probably is sufficient, and likely just the "user" component. Better (apparently!) is adding user and CPU, and
using gcFirst=TRUE argument. 

Also Arkajyoti and I last year looked into identifying the machine that did the work. See MachID.R in 
https://github.com/nashjc/RNonlinearLS/tree/main/MachineSummary

I've not yet any solid opinion WHERE to do the timing. Probably surrounding the local_wrapper I think, so that we reflect the setup costs. However, it could be argued we want to just time the call to the solver function.

JN

**Comment by SudoWodo** [comment]
Got that, I'll try timing the solver function inside the local wrapper.

Do we have to output the time for both global_wrapper and sopm?

**Comment by nashjc** [comment]
In opm() I put the time as one column of the table of outputs so they can be compared. They aren't a "good" timing, but an indication.
Timings are very sensitive to other activity in a computer. And, of course, to the computer resources.
